app/ide-desktop/lib/client/src/index.ts

```diff
@@ -0,0 +1,335 @@
+// FIXME: Issues to be resolved:
+// - Enable vibrancy on MacOS
+
+import Electron from 'electron'
+import isDev from 'electron-is-dev'
+import path from 'node:path'
+import * as server from 'bin/server'
+import * as content from '../../content/src/config'
+import * as security from 'security'
+import * as naming from 'naming'
+import * as config from 'config'
+import * as paths from 'paths'
+import * as configParser from 'config/parser'
+import * as debug from 'debug'
+import * as projectManager from 'bin/project-manager'
+
+const logger = content.logger
+
+// ============
+// === Help ===
+// ============
+
+/** The Electron application. It is responsible for starting all the required services, and
+ * displaying and managing the app window. */
+class App {
+    window: null | Electron.BrowserWindow = null
+    server: null | server.Server = null
+    args: config.Args
+    windowSize: config.WindowSize
```

Comment:
Isn't this already a part on the `window`?
Does this have any meaning if the `window` is `null`?